arcgis-ios-sdk-samples/Utility network/Perform valve isolation trace/PerformValveIsolationTraceViewController.swift

```diff
@@ -260,25 +351,126 @@ class PerformValveIsolationTraceViewController: UIViewController {
             preferredStyle: .actionSheet
         )
         filterBarrierCategories.forEach { category in
-            let action = UIAlertAction(title: category.name, style: .default) { _ in
-                self.selectedCategory = category
-                self.setStatus(message: "\(category.name) selected.")
+            let action = UIAlertAction(title: category.name, style: .default) { [self] _ in
+                selectedCategory = category
+                setStatus(message: "\(category.name) selected.")
+                traceResetBarButtonItem.isEnabled = true
             }
             alertController.addAction(action)
         }
         let cancelAction = UIAlertAction(title: "Cancel", style: .cancel)
         alertController.addAction(cancelAction)
-        alertController.popoverPresentationController?.barButtonItem = categoryButton
+        alertController.popoverPresentationController?.barButtonItem = categoryBarButtonItem
         present(alertController, animated: true)
     }
     
     // MARK: UIViewController
     
     override func viewDidLoad() {
         super.viewDidLoad()
-        
-        loadServiceGeodatabase()
         // Add the source code button item to the right of navigation bar.
-        (self.navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["PerformValveIsolationTraceViewController"]
+        (navigationItem.rightBarButtonItem as? SourceCodeBarButtonItem)?.filenames = ["PerformValveIsolationTraceViewController"]
+        // Load the service geodatabase and utility network.
+        setStatus(message: "Loading utility network…")
+        loadServiceGeodatabase()
+    }
+}
+
+// MARK: - AGSGeoViewTouchDelegate
+
+extension PerformValveIsolationTraceViewController: AGSGeoViewTouchDelegate {
+    func geoView(_ geoView: AGSGeoView, didTapAtScreenPoint screenPoint: CGPoint, mapPoint: AGSPoint) {
+        if let identifyAction = identifyAction {
+            identifyAction.cancel()
+        }
+        identifyAction = mapView.identifyLayers(atScreenPoint: screenPoint, tolerance: 10, returnPopupsOnly: false) { [weak self] result, error in
```

Comment:
🤔  Not sure if we should block user interaction here, as we are identifying features in online layers (which could be not responsive when the connection is bad)

Reply:
That sounds like a good idea.